src/main/scala/com/microsoft/hyperspace/index/DataFrameWriterExtensions.scala

```diff
@@ -50,7 +50,8 @@ object DataFrameWriterExtensions {
         df: DataFrame,
         path: String,
         numBuckets: Int,
-        bucketByColNames: Seq[String]): Unit = {
+        bucketByColNames: Seq[String],
+        mode: SaveMode = SaveMode.Overwrite): Unit = {
```

Comment:
If this function is being changed to support other modes, lets add a quick test case for `SaveMode.Append` under `DataFrameWriterExtensionsTests`.